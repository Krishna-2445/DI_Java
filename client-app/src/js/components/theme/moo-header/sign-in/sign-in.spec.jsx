import React from 'react';
import { shallow } from 'enzyme';
import SignIn from './sign-in';

describe('<SignIn>', () => {
  describe('with no authenticated User', () => {
    it('renders Sign In with link from props', () => {
      const wrapper = shallow(<SignIn link="/sign-in" />);
      expect(wrapper.contains('Sign In')).toEqual(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with an authenticated user', () => {
    it('should show user name and default link to nothing', () => {
      const user = {
        firstName: 'Abby',
        lastName: 'Jones',
      };
      const wrapper = shallow(<SignIn authenticatedUser={user} />);
      expect(wrapper.contains('Abby Jones')).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it('should use alternate link if provided', () => {
      const user = {
        firstName: 'Abby',
        lastName: 'Jones',
        link: '/abby-jones',
      };
      const wrapper = shallow(<SignIn authenticatedUser={user} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
