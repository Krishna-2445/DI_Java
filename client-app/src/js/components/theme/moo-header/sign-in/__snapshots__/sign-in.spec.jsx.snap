// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn> with an authenticated user should show user name and default link to nothing 1`] = `
<MenuItem
  activeClassName="active"
  as={[Function]}
  id="moo-sign-in"
  to="#"
>
  <Icon
    as="i"
    name="user circle outline"
  />
  Abby Jones
</MenuItem>
`;

exports[`<SignIn> with an authenticated user should use alternate link if provided 1`] = `
<MenuItem
  activeClassName="active"
  as={[Function]}
  id="moo-sign-in"
  to="/abby-jones"
>
  <Icon
    as="i"
    name="user circle outline"
  />
  Abby Jones
</MenuItem>
`;

exports[`<SignIn> with no authenticated User renders Sign In with link from props 1`] = `
<MenuItem
  activeClassName="active"
  as={[Function]}
  id="moo-sign-in"
  to="/sign-in"
>
  <Icon
    as="i"
    name="user circle outline"
  />
  Sign In
</MenuItem>
`;
